students/arkadiusz_kasprzyk/lesson_06_dicts/cubes.py

```diff
@@ -0,0 +1,43 @@
+# cubes.py
+
+
+def cubes(aset, bset):
+    sets_inter = aset & bset
+    aset_only = aset - bset
+    bset_only = bset - aset
+    return sets_inter, aset_only, bset_only
+
+
+def into_set(n=0, aset=set()):
+    """
+    Examples
+    --------
+    into_set()
+    into_set(2)
+    """
+    print("Enter elements of the set (only integers!): ")
+    for k in range(n):
+        aset |= {int(input("{}: ".format(k + 1)))}
+    print(aset)
+    return aset
+
+
+def print_set(aset):
+    print("{} elements in the set".format(len(aset)))
+    ll = list(aset)
+    ll.sort()
+    for e in ll:
+        print(e)
+
+
+if __name__ == "__main__":
+
+    nums = []
+    while len(nums) < 2:
+        nums += input("Give numbers of elements in sets: ").split(" ")
+
+    nums = nums[:2]
+    nums = [int(n) for n in nums]
+
+    for s in cubes(into_set(nums[0], set()), into_set(nums[1], set())):
```

Comment:
This is a matter of what is *considered* to be more basic data structure, list or dictionary. Finally I think it doesn't matter: I may have method for reading into list so that I can also have a method reading *directly* into dictionary, i.e. methods reading the input and returning directly the type I wish to obtain: list or dictionary (or whatever). I don't feel like lists should be anyhow distinguished and created only to be immediately converted into set (notice that into_set() doesn't internally create any list).
After all, I agree that the solution is not quite clear 
but I simply applied DRY for the purpose of this specific script.

Reply:
Yes, the name is not good... However, my idea was to read input into an already existing set (and by default it is an empty set) - now I see it's not reasonable as proper set methods already exist. Silly me... 
Except this one generalisation the function was just a product of DRY for the purpose of this specific script with no intention to make any further use of it - so it's just a bag for all the things I need at this place. Ugly but fast. The way it takes data results from the form of tests in Snakify.